### ISSUE TITLE
Upgrade GCC from 8.1.0 to 8.2.0 fails with brew

### ISSUE THREAD

**Issue by Moonbloom** [Homebrew/homebrew-core#30655]
Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.

 have a problem with brew install (or upgrade, reinstall) a single formula? If it's a general brew problem please file this issue at Linuxbrew/brew: https://github.com/Linuxbrew/brew/issues/new/choose. If it's a tap (e.g. Brewsci/homebrew-bio) problem please file this issue at the tap.
 ran brew update and can still reproduce the problem?
 ran brew doctor, fixed all issues and can still reproduce the problem?
 ran brew gist-logs <formula> (where <formula> is the name of the formula that failed) and included the output link?
 if brew gist-logs didn't work: ran brew config and brew doctor and included their output with your issue?
To help us debug your issue please explain:

What you were trying to do (and why)
I'm trying to upgrade GCC from 8.1.0 to 8.2.0.
I always try to keep everything up to date.

What happened (include command output)
When trying to upgrade GCC, i run into the following issue:

09:23:51 Kaspers-MacBook-Pro:~ Kasper$ brew upgrade
==> Upgrading 1 outdated package, with result:
gcc 8.1.0 -> 8.2.0
==> Upgrading gcc
Warning: Building gcc from source:
The bottle needs the Xcode CLT to be installed.
==> Downloading https://ftp.gnu.org/gnu/gcc/gcc-8.2.0/gcc-8.2.0.tar.xz
Already downloaded: /Users/Kasper/Library/Caches/Homebrew/gcc-8.2.0.tar.xz
==> ../configure --build=x86_64-apple-darwin17.7.0 --prefix=/usr/local/Cellar/gcc/8.2.0 --libdir=/us
==> make
Last 15 lines from /Users/Kasper/Library/Logs/Homebrew/gcc/02.make:
__isl_give isl_set *isl_map_domain(__isl_take isl_map *bmap);
^
../../gcc/graphite-sese-to-poly.c:524:26: error: cannot initialize a parameter of type 'isl_map *' with an rvalue of type 'isl_space *'
isl_space_domain (isl_map_get_space (acc)));
^~~~~~~~~~~~~~~~~~~~~~~
/usr/local/opt/isl/include/isl/map.h:496:56: note: passing argument to parameter 'bmap' here
__isl_give isl_set *isl_map_domain(__isl_take isl_map *bmap);
^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[3]: *** [graphite-sese-to-poly.o] Error 1
rm gcc.pod
make[2]: *** [all-stage1-gcc] Error 2
make[1]: *** [stage1-bubble] Error 2
make: *** [all] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting

What you expected to happen
GCC to correctly run it's make file and finish correctly.

Step-by-step reproduction instructions (by running brew install commands)
brew update
brew upgrade

gist-logs:
https://gist.github.com/a69b0580dd740262b560f08cd35265c6

brew doctor only outputs 'Your system is ready to brew.'

I have previously been able to update GCC to versions below this (for example 8.0.0 -> 8.1.0)

**Comment by DomT4** [comment]
Yeah it looks like the new `isl` release broke `gcc`. That's fun. Upstream has [another](https://gcc.gnu.org/ml/gcc-bugs/2018-07/msg02547.html) bug report for `isl` already and it's not even the same issue, seemingly.

**Comment by DomT4** [comment]
https://github.com/Homebrew/homebrew-core/pull/30639 fixed one `isl`/`gcc` issue, hopefully this one. Make sure you've ran `brew update` & try again.
